### PR TITLE
feat: add site award progress type helper

### PR DESCRIPTION
**Pull request by luchaos**
Fixes tiered awards reading duplicates for SiteAwards.

### CODE REVIEW

File: app/Helpers/database/site-award.php [hunk 1]
```diff
@@ -92,18 +92,26 @@ function getUsersSiteAwards(string $user, bool $showHidden = false): array
...
-                  LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
-                  WHERE (saw.AwardType = " . AwardType::Mastery . " OR saw.AwardType = " . AwardType::GameBeaten . ") AND saw.User = :username
-                  GROUP BY saw.AwardType, saw.AwardData, saw.AwardDataExtra
-    UNION
-    SELECT " . unixTimestampStatement('MAX(saw.AwardDate)', 'AwardedAt') . ", saw.AwardType, MAX( saw.AwardData ), saw.AwardDataExtra, saw.DisplayOrder, NULL, NULL, NULL, NULL, NULL
-                  FROM SiteAwards AS saw
-                  WHERE saw.AwardType > " . AwardType::Mastery . " AND saw.User = :username2
-                  GROUP BY saw.AwardType
-    ORDER BY DisplayOrder, AwardedAt, AwardType, AwardDataExtra ASC";
+        -- non-tiered awards (mastery, beaten)
```
Comment (Jamiras) [inline app/Helpers/database/site-award.php:95]: These are actually Game Awards (note the join against GameData). The other group is non-game awards, which does include the tiered awards, but also includes the patreon and certified legends awards.

File: app/Helpers/database/site-award.php [hunk 2]
```diff
@@ -92,18 +92,26 @@ function getUsersSiteAwards(string $user, bool $showHidden = false): array
...
+        -- tiered awards (developer contribution, ...)
+        SELECT " . unixTimestampStatement('MAX(saw.AwardDate)', 'AwardedAt') . ", saw.AwardType, MAX( saw.AwardData ), saw.AwardDataExtra, saw.DisplayOrder, NULL, NULL, NULL, NULL, NULL
+            FROM SiteAwards AS saw
+            WHERE
+                saw.AwardType NOT IN(" . implode(',', AwardType::progress()) . ")
+                AND saw.User = :username2
+            GROUP BY saw.AwardType
+        ORDER BY DisplayOrder, AwardedAt, AwardType, AwardDataExtra ASC";
+
+    ray($query);
```
Comment (Jamiras) [inline app/Helpers/database/site-award.php:114]: What is `ray`?

Comment (luchaos) [reply app/Helpers/database/site-award.php]: https://myray.app/
i use it for debugging sometimes

File: app/Community/Enums/AwardType.php [hunk 3]
```diff
@@ -53,4 +53,17 @@ public static function toString(int $awardType): string
...
+    {
+        return [
+            self::Mastery,
+            self::GameBeaten,
+        ];
+    }
+
+    public static function isGame(int $type): bool
+    {
+        return !in_array($type, static::game());
```
Comment (Jamiras) [inline app/Community/Enums/AwardType.php:67]: should be `in_array`:
image

Comment (luchaos) [reply app/Community/Enums/AwardType.php]: yup, tests didn't like that either - nice.
fixed in https://github.com/RetroAchievements/RAWeb/pull/1959/commits/327b3d67e53ff5fc26ae16b45d98f84c9d529669

**Review by Jamiras — APPROVED** [review]

**Pull request merged**